assets/app/scripts/controllers/pod.js

```diff
@@ -79,10 +81,35 @@ angular.module('openshiftConsole')
       watches.push(DataService.watch("builds", $scope, function(builds) {
         $scope.builds = builds.by("metadata.name");
         Logger.log("builds (subscribe)", $scope.builds);
-      }));      
+      }));
+
+      (function initLogs() {
+        angular.extend($scope, {
+          logs: [],
+          logsLoading: true,
+          canShowDownload: false,
+          initLogs: initLogs
+        });
+        DataService.watchLogStream("pods/log", $routeParams.pod, $scope, function(line) {
+          $scope.$apply(function() {
+            $scope.logs.push({text: line});
+            $scope.canShowDownload = true;
+          });
+        }, function() {
+          $scope.$apply(function() {
+            $scope.logsLoading = false;
+          });
+        })
```

Comment:
Ok, I have an alternate version of this streamer implemented.  Pasting both here for comparison (my actual work is in a different branch in case I get a :rage: :frowning: or :confused:):  

```javascript

        // VERSION 2 OF BUILDING A STREAMER
       // ----------------------------------------------------------
        var streamer = DataService.stream('pods/log',$routeParams.pod, $scope);
        streamer.open(function(evt) {
          console.log('OPENING', evt);
        });
        streamer.message(function(msg) {
          $scope.$apply(function() {
            $scope.logs.push({text: msg});
            $scope.canShowDownload = true;
          });
        });
        // raw data callback version
        streamer.message(function(msg) {
          console.log('RAW looks neat:', msg);
        }, true);
        streamer.close(function(evt) {
          console.log('CLOSING', 'its closed.', evt);
        });
        streamer.error(function(evt) {
          console.log('ERROR', 'sad face.', evt);
        });

        streamer
          .go()
          .then(function(handle) {
            $scope.$on("$destroy", function() {
              DataService.unStream(handle);
            });
          });

        // THE FIRST VERSION, FOR COMPARISON:
       // ----------------------------------------------------------
        DataService.watchLogStream("pods/log", $routeParams.pod, $scope, function(line) {
          $scope.$apply(function() {
            $scope.logs.push({text: line});
            $scope.canShowDownload = true;
          });
        }, function() {
          $scope.$apply(function() {
            $scope.logsLoading = false;
          });
        })
        .then(function(ws) {
          $scope.$on("$destroy", function() {
            DataService.unwatchLogStream(ws);
          });
        });

```

@spadgett @liggitt @jwforres thoughts on how we want this to look....